### PR TITLE
Add tray support

### LINKED ISSUE #864: SlimeVR-Server should have the ability to close to tray
>> Suggested solution or mitigation of the issue:
>> SlimeVR-Server should minimize to tray on "close" of the app so that trackers can keep communicating with the server. Shut down of the app should only be by right-clicking on the tray icon of SlimeVR-Server and selecting "exit". By applying this change the issue is mitigated as shut down of the server cannot happen by mistake.
>
>Sorry, this is clearly a wrong solution :D The issue should be fixed. If there is a request for server to be minimized on close, the issue should be created for this specifically unrelated to the bug with the trackers.
>
>_Originally posted by @Eirenliel in https://github.com/SlimeVR/SlimeVR-Server/issues/863#issuecomment-1743350400_
           
As mentioned in a reply by Eirenliel I am raising another issue for the ability to close the application to tray.
Similarly to many other applications, the app should be able to minimize to tray. The behaviour can also be configurable in settings.

The reason why I think "close to tray" would be useful is that the trackers need the server running to work. If default closure behaviour would send the app to tray then it can keep working in the background.

As another alternative on click of "close" a dialog could be opened giving three options: minimize to tray, close app, back

### PR DESCRIPTION
**Pull request by ImUrX**
Fixes #864 

In the first time being closed, it will ask you if you want to save to tray or close SlimeVR
modal image of first close

### CODE REVIEW

File: gui/public/i18n/en/translation.ftl [hunk 1]
```diff
@@ -421,6 +421,9 @@ settings-general-interface-feedback_sound-volume = Feedback sound volume
 settings-general-interface-connected_trackers_warning = Connected trackers warning
 settings-general-interface-connected_trackers_warning-description = This option will show a pop-up every time you try exiting SlimeVR while having one or more connected trackers. It reminds you to turn off your trackers when you are done to preserve battery life.
 settings-general-interface-connected_trackers_warning-label = Connected trackers warning on exit
+settings-general-interface-use_tray = Close to system tray
+settings-general-interface-use_tray-description = Let's you close the window without stopping the SlimeVR server so you can continue using it without having the GUI bothering you.
```
Comment (Louka3000) [inline gui/public/i18n/en/translation.ftl:425]:
```suggestion
settings-general-interface-use_tray-description = Lets you close the window without closing the SlimeVR Server so you can continue using it without having the GUI bothering you.
```

File: gui/public/i18n/en/translation.ftl [hunk 2]
```diff
@@ -421,6 +421,9 @@ settings-general-interface-feedback_sound-volume = Feedback sound volume
 settings-general-interface-connected_trackers_warning = Connected trackers warning
 settings-general-interface-connected_trackers_warning-description = This option will show a pop-up every time you try exiting SlimeVR while having one or more connected trackers. It reminds you to turn off your trackers when you are done to preserve battery life.
 settings-general-interface-connected_trackers_warning-label = Connected trackers warning on exit
+settings-general-interface-use_tray = Close to system tray
```
Comment (Louka3000) [inline gui/public/i18n/en/translation.ftl:424]:
```suggestion
settings-general-interface-use_tray = Minimize to system tray
```

File: gui/public/i18n/en/translation.ftl [hunk 3]
```diff
@@ -421,6 +421,9 @@ settings-general-interface-feedback_sound-volume = Feedback sound volume
...
+settings-general-interface-use_tray-description = Let's you close the window without stopping the SlimeVR server so you can continue using it without having the GUI bothering you.
+settings-general-interface-use_tray-label = Close to system tray
```
Comment (Louka3000) [inline gui/public/i18n/en/translation.ftl:426]:
```suggestion
settings-general-interface-use_tray-label = Minimize to system tray
```

File: gui/public/i18n/en/translation.ftl [hunk 4]
```diff
@@ -865,3 +868,19 @@ status_system-StatusSteamVRDisconnected = { $type ->
...
+
+## Tray Menu
+tray_menu-show = Show
+tray_menu-hide = Hide
+tray_menu-quit = Quit
+
+## First exit modal
+tray_or_exit_modal-title = What should the close button do?
+# Multiline text
+tray_or_exit_modal-description = This let's you choose if you want to exit or close to tray when pressing the close button.
```
Comment (Louka3000) [inline gui/public/i18n/en/translation.ftl:880]:
```suggestion
tray_or_exit_modal-description = This lets you choose whether you want to close the server or to minimize it to the tray when pressing the close button.
```

File: gui/public/i18n/en/translation.ftl [hunk 5]
```diff
@@ -865,3 +868,19 @@ status_system-StatusSteamVRDisconnected = { $type ->
...
+
+    You can later change this in the interface settings!
```
Comment (Louka3000) [inline gui/public/i18n/en/translation.ftl:882]:
```suggestion
    You can change this later in the interface settings!
```

**Review by Louka3000 — CHANGES_REQUESTED** [review]

**Review by Louka3000 — APPROVED** [review]

**Review by loucass003 — APPROVED** [review]

**Review by ButterscotchV — APPROVED** [review]
Looks fine, can't review the Rust code thoroughly.

**Pull request merged**